File: server/router/shop/shop_analytics.go
package shop

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/gin-gonic/gin"
)

type ShopAnalyticsRouter struct{}

func (s *ShopAnalyticsRouter) InitShopAnalyticsRouter(Router *gin.RouterGroup) {
	shopAnalyticsRouter := Router.Group("shopAnalytics")
	var api = v1.ApiGroupApp.ShopApiGroup.ShopAnalyticsApi
	{
		shopAnalyticsRouter.GET("getDashboardOverview", api.GetDashboardOverview)
		shopAnalyticsRouter.GET("getSalesReport", api.GetSalesReport)
		shopAnalyticsRouter.GET("getTopProducts", api.GetTopProducts)
		shopAnalyticsRouter.GET("getCategorySales", api.GetCategorySales)
		shopAnalyticsRouter.GET("getOrderStatusDistribution", api.GetOrderStatusDistribution)
	}
}

File: web/src/api/shopAnalytics.js
import service from '@/utils/request'

export const getDashboardOverview = (params) => {
  return service({ url: '/shopAnalytics/getDashboardOverview', method: 'get', params })
}

export const getSalesReport = (params) => {
  return service({ url: '/shopAnalytics/getSalesReport', method: 'get', params })
}

export const getTopProducts = (params) => {
  return service({ url: '/shopAnalytics/getTopProducts', method: 'get', params })
}

export const getCategorySales = (params) => {
  return service({ url: '/shopAnalytics/getCategorySales', method: 'get', params })
}

export const getOrderStatusDistribution = (params) => {
  return service({ url: '/shopAnalytics/getOrderStatusDistribution', method: 'get', params })
}
